aws/client/client.go

```diff
@@ -0,0 +1,34 @@
+package client
+
+import (
+	"context"
+
+	"github.com/awslabs/smithy-go/middleware"
+	"github.com/awslabs/smithy-go/transport/http"
+)
+
+var _ http.Header
+var _ middleware.MiddlewareHandler
+
+type signingMetadataKey struct{}
+
+type SigningMetadata struct {
+	SigningName   string
+	SigningRegion string
+}
+
+func GetSigningMetadata(ctx context.Context) SigningMetadata {
```

Comment:
Should this signing metadata be apart of a base client package? This seems more specific to a signer package, that a generated client would compose it self with, or have members for.

Reply:
Agreed, this should probably be moved to the v4 package.